beginner_source/blitz/tensor_tutorial.py

```diff
@@ -1,195 +1,166 @@
-# -*- coding: utf-8 -*-
 """
-What is PyTorch?
-================
+Tensors - The building blocks of deep learning
+--------------------------------------------
 
-It’s a Python-based scientific computing package targeted at two sets of
-audiences:
+Tensors are a specialized data structure that are very similar to arrays
+and matrices. In PyTorch, we use tensors to encode the inputs and
+outputs of a model, as well as the model’s parameters.
 
--  A replacement for NumPy to use the power of GPUs
--  a deep learning research platform that provides maximum flexibility
-   and speed
+Tensors are similar to NumPy’s ndarrays, except that tensors can run on
+a GPU to accelerate computing. If you’re familiar with ndarrays, you’ll
+be right at home with the Tensor API. If not, follow along in this quick
+API walkthrough.
 
-Getting Started
----------------
-
-Tensors
-^^^^^^^
-
-Tensors are similar to NumPy’s ndarrays, with the addition being that
-Tensors can also be used on a GPU to accelerate computing.
 """
 
-from __future__ import print_function
 import torch
+import numpy as np
 
-###############################################################
-# .. note::
-#     An uninitialized matrix is declared,
-#     but does not contain definite known
-#     values before it is used. When an
-#     uninitialized matrix is created,
-#     whatever values were in the allocated
-#     memory at the time will appear as the initial values.
 
-###############################################################
-# Construct a 5x3 matrix, uninitialized:
+######################################################################
+# Tensor Initialization
+# ~~~~~~~~~~~~~~~~~~~~~
+#
+# Tensors can be initialized in various ways. Take a look at the following examples
+#
+# **Directly from data / NumPy arrays:**
+#
 
-x = torch.empty(5, 3)
-print(x)
- 
-###############################################################
-# Construct a randomly initialized matrix:
+data = [[1, 2],[3, 4]]
+np_array = np.array(data)
 
-x = torch.rand(5, 3)
-print(x)
+tnsr_from_data = torch.tensor(data)
```

Comment:
I think here you want to mention that the types are inferred either from the data or from the numpy array (will echo later in the error you try to show).